1.1-Introduction-to-Schnorr.ipynb

```diff
@@ -230,35 +239,45 @@
    "outputs": [],
    "source": [
     "msg = hashlib.sha256(b'message').digest()\n",
-    "d = ECKey().generate()\n",
-    "P = d.get_pubkey()\n",
-    "print(\"message = {}\".format(msg))\n",
-    "print(\"pubkey = {}\".format(P.get_bytes()))\n",
+    "d, P = generate_key_pair()\n",
+    "print(\"message = {}\".format(msg.hex()))\n",
+    "print(\"pubkey = {}\\n\".format(P.get_bytes().hex()))\n",
     "\n",
     "# Generate the nonce value k deterministically and get the nonce point R\n",
     "# Method: use hashlib.sha256(bytes).digest() on the message and pubkey\n",
-    "k =  # TODO: implement\n",
-    "R =  # TODO: implement\n",
+    "k_bytes =  # TODO: implement\n",
+    "k, R =  # TODO: implement\n",
     "\n",
     "# Check that nonce is quadratic residue modulo the field size\n",
     "# If not, negate k\n",
-    "# Method: jacobi_symbol(R.get_y(), SECP256K1_FIELD_SIZE)\n",
+    "# Method: jacobi_symbol(int(y(R)), SECP256K1_FIELD_SIZE)\n",
     "if  # TODO: implement\n",
     "    k.negate()\n",
     "\n",
-    "# Generate s = k + H(R_x|P|msg) * d\n",
+    "print(\"nonce: {}\".format(k))\n",
+    "print(\"nonce point: {}\\n\".format(R))\n",
+    "\n",
+    "# Generate s = k + sha256(R_x|P|msg) * d\n",
     "# Method: hashlib.sha256(bytes).digest() will give you the byte digest of the SHA256 of some bytes\n",
     "# Turn that digest into a ECKey object called h, and then set s = k + h * d\n",
+    "R_x_bytes = R.get_x().to_bytes(32, 'big')\n",
+    "P_bytes = P.get_bytes()\n",
     "h_bytes =  # TODO: implement\n",
-    "h =  # TODO: implement\n",
+    "h = ECKey().set(h_bytes)\n",
     "s = k + h * d\n",
     "\n",
+    "print(\"R: {}\".format(R.get_bytes().hex()))\n",
+    "print(\"s: {}\\n\".format(s.secret.to_bytes(32, 'big').hex()))\n",
```

Comment:
you can just use `get_bytes()` on an ECKey

Reply:
Yes. That's better. Thanks!